src/NUnitFramework/framework/Internal/Filters/AndFilter.cs

```diff
@@ -54,9 +79,13 @@ public override bool Match( ITest test )
         /// <returns>True if all the component filters explicit match, otherwise false</returns>
         public override bool IsExplicitMatch( ITest test )
         {
-            foreach( TestFilter filter in Filters )
-                if ( !filter.IsExplicitMatch( test ) )
+            foreach (var filter in Filters)
```

Comment:
There are 4 nearly identical loops here.  Can that be simplified?

Reply:
I think that might become hard to read if we try parametrize the logic, all the Pass == false -> return true inverting etc